File: test_data/csharp/test_main/aas_core_meta.v3rc2/input/snippets/Verification/data_specification_IEC_61360s_have_definition_at_least_in_english.cs
/// <summary>
/// Check that the <see cref="Aas.IDataSpecificationIec61360.Definition" /> is defined
/// for all data specifications whose content is given as IEC 61360 at least in English.
/// </summary>
[CodeAnalysis.SuppressMessage("ReSharper", "InconsistentNaming")]
public static bool DataSpecificationIec61360sHaveDefinitionAtLeastInEnglish(
    IEnumerable<Aas.IEmbeddedDataSpecification> embeddedDataSpecifications
)
{
    foreach (var embeddedDataSpecification in embeddedDataSpecifications)
    {
        var iec61360 = (
            embeddedDataSpecification.DataSpecificationContent
                as Aas.IDataSpecificationIec61360
        );
        if (iec61360 != null)
        {
            if (iec61360.Definition == null)
            {
                return false;
            }

            var noDefinitionInEnglish = true;
            foreach (var langString in iec61360.Definition)
            {
                if (IsBcp47ForEnglish(langString.Language))
                {
                    noDefinitionInEnglish = false;
                    break;
                }
            }

            if (noDefinitionInEnglish)
            {
                return false;
            }
        }
    }

    return true;
}

File: test_data/typescript/test_main/aas_core_meta.v3rc2/input/snippets/Verification/data_specification_IEC_61360s_have_definition_at_least_in_english.ts
/**
 * Check that {@link types.DataSpecificationIec61360.definition}
 * is defined for all data specifications whose content is given as
 * IEC 61360 at least in English.
 *
 * @param embeddedDataSpecifications - to be verified
 * @returns `true` if the check passes
 */
export function dataSpecificationIec61360sHaveDefinitionAtLeastInEnglish(
  embeddedDataSpecifications: Iterable<AasTypes.EmbeddedDataSpecification>
): boolean {
  for (const embeddedDataSpecification of embeddedDataSpecifications) {
    const content = embeddedDataSpecification.dataSpecificationContent;
    if (AasTypes.isDataSpecificationIec61360(content)) {
      if (content.definition === null) {
        return false;
      }

      let noDefinitionInEnglish = true;
      for (const langString of content.definition) {
        if (isBcp47ForEnglish(langString.language)) {
          noDefinitionInEnglish = false;
          break;
        }
      }

      if (noDefinitionInEnglish === true) {
        return false;
      }
    }
  }

  return true;
}
